src/hotspot/share/jfr/support/jfrThreadLocal.hpp

```diff
@@ -79,6 +85,25 @@ class JfrThreadLocal {
   bool _dead;
   bool _sampling_critical_section;
 
+#ifdef LINUX
+  bool _has_cpu_timer = false;
+  timer_t _cpu_timer;
+
+  enum CPUTimeLockState {
+    UNLOCKED,
+    // locked for enqueuing
+    ENQUEUE,
+    // locked for dequeuing
+    DEQUEUE,
+    // locked for writing native event out of safepoint
+    NATIVE
+  };
+  volatile CPUTimeLockState _cpu_time_jfr_locked = UNLOCKED;
+  volatile bool _has_cpu_time_jfr_requests = false;
+  JfrCPUTimeTraceQueue _cpu_time_jfr_queue{0};
```

Comment:
Please initialize these fields in the initializer list, like the other fields.

Also, what capacity will the JfrCPUTimeTraceQueue have with an initializer like the above - {0}? In GDB i see that the capacity is 10000. Each element is 48 bytes, meaning each thread now has an associated queue of 468 kb, almost 0.5 Mb. Is that correct?

Reply:
I see that JfrCPUTimeTraceQueue::ensure_capacity_for_period() calculates and sets the queue size as a function of the sampling period (in my case its 1 ms).